auth-table-based-service/src/main/java/io/stargate/auth/table/AuthnTableBasedService.java

```diff
@@ -220,60 +222,75 @@ protected static boolean checkpw(String password, String hash) {
 
   @Override
   public StoredCredentials validateToken(String token) throws UnauthorizedException {
+    UUID uuid = parseToken(token);
+
+    try {
+      ResultSet resultSet = queryAuthTable(uuid).get();
+      Row row = extractCredentialsRow(resultSet);
+      updateAuthTableTtl(row).get();
+      return new StoredCredentials().roleName(row.getString("username"));
+    } catch (InterruptedException | ExecutionException e) {
+      logger.error("Failed to validate token", e);
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  public CompletionStage<StoredCredentials> validateTokenAsync(String token) {
+    return CompletableFuture.completedFuture(token)
+        .thenApply(this::parseToken)
+        .thenCompose(this::queryAuthTable)
+        .thenApply(this::extractCredentialsRow)
+        .thenCompose(
+            row ->
+                updateAuthTableTtl(row)
+                    .thenApply(__ -> new StoredCredentials().roleName(row.getString("username"))));
```

Comment:
We wait for the TTL update to complete before returning the response. I did it that way to mimic the behavior of the sync version, but I think we could fire the update asynchronously and not wait for the response, it doesn't really impact the validation.

I can make that change in both methods (sync and async).